### PR TITLE
microswitch pod for V2.4 with JST-XH 4-pin connector mount

### PR DESCRIPTION
**Pull request by BladeScraper-Designs**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?
Adds a JST-XH 4-pin connector to the microswitch pod to allow easy disconnection for any work you need to do, whether general disassembly, maintenance, repair of the switches, etc. 
#### How was it tested? 
On my own 2.4
#### Further notes
as stated in the readme, because the part involves bridging one may have to mess around a bit with getting the JST fitment properly.  Might have to take a knife to it to cut off and sagging bridges.

### CODE REVIEW

**Comment by BladeScraper-Designs** [comment]
Added another mod to my repository.

**Comment by BladeScraper-Designs** [comment]
wasn't sure if I had to do another pull request or what.  I'm new to GitHub :)
https://github.com/BladeScraper-Designs/VoronUsers/tree/master/printer_mods/BladeScraper-Designs/2020_PiCamV2_Mount

**Comment by BladeScraper-Designs** [comment]
Learned how to properly do multiple mods at once, as advised by ShiftingTech on Discord.  Creating new branches for new mods.

File: printer_mods/BladeScraper-Designs/Microswitch_Pod_JST/readme.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+This is a simple modification to the microswitch pod to allow the installation of a JST-XH 4-pin connector.
+
+JST-XH  4-pin connector part number B4B-XH-A(LF)(SN) slides into the hole and there's a small lip to keep it sliding all the way thru.  
+Depending on your tolerances it may be tight enough as is, or you may have to use a bit of CA glue.  You may need to clean up the inside of the slot a bit using a utility knife to get the JST in, as it does involve bridging.
+
+I also slightly increased the size of the screw holes from 1.5mm to 1.7mm as I was having a hard time getting mine screwed down.  
+
+![alt text](https://github.com/BladeScraper-Designs/VoronUsers/blob/master/printer_mods/BladeScraper-Designs/Microswitch_Pod_JST/img/assembly.png?raw=true)
```
Comment (KiloQubit) [inline printer_mods/BladeScraper-Designs/Microswitch_Pod_JST/readme.md:8]: Please update the URL to be relative.  Something like

    Microswitch Pod with Connector

File: printer_mods/README.md [hunk 2]
```diff
@@ -138,5 +138,6 @@ like so:
 | | [V0 Electronics Cover](./xbst_/V0_Electronic_Panel)| A printed cover for v0 electronics. Requires larger printer to print. | :heavy_check_mark: | :x: | :x: | :x: |
 | | [V0 PiCam Mount](./xbst_/V0_Picam)| A Raspberry Pi Camera Mount For V0 | :heavy_check_mark: | :x: | :x: | :x: |
 | | [DuetZero](./xbst_/DuetZero) | Pi Zero mount over a Duet2 for space saving | :heavy_check_mark: |:heavy_check_mark: |:heavy_check_mark: |:x:|
+| BladeScraper-Designs | [Microswitch-Pod-JST](./BladeScraper-Designs/Microswitch_Pod_JST) | Microswitch pod with JST-XH 4-pin connector | :x: |:x: |:heavy_check_mark: | :x:
```
Comment (KiloQubit) [inline printer_mods/README.md:141]: Please move this line up so that it's in alphanumeric order by username, not at the end of the file.

**Review by KiloQubit — CHANGES_REQUESTED** [review]

File: printer_mods/BladeScraper-Designs/Microswitch_Pod_JST/readme.md (hunk 1 above)
Comment (BladeScraper-Designs) [reply printer_mods/BladeScraper-Designs/Microswitch_Pod_JST/readme.md]: Done, I think lol.  Not great at using GH.  Lmk if I did it right.

File: printer_mods/README.md (hunk 2 above)
Comment (BladeScraper-Designs) [reply printer_mods/README.md]: I could have sworn I did but must have been on a different branch.  Fixed.

**Review by KiloQubit — APPROVED** [review]

**Review by KiloQubit — APPROVED** [review]

**Review by FHeilmann — APPROVED** [review]

**Pull request merged**